narwhals/_compliant/dataframe.py

```diff
@@ -270,6 +272,9 @@ def _evaluate_aliases(self, *exprs: CompliantExprT_contra) -> list[str]:
         it = (expr._evaluate_aliases(self) for expr in exprs)
         return list(chain.from_iterable(it))
 
+    def _check_columns_exist(self, subset: Sequence[str]) -> ColumnNotFoundError | None:
+        return check_columns_exist(subset, available=list(self.columns))
```

Comment:
added this method at compliant level as @dangotbanned suggested.

I like it way more now

Reply:
great point and awesome explanation. updated :)